shm/Ship.hpp

```diff
@@ -30,11 +34,15 @@ class Ship {
     Ship& operator-=(size_t num);
     Ship& operator+=(size_t num);
 
+    void load(const std::shared_ptr<Cargo>& cargo) { allCargos_.push_back(cargo); }
```

Comment:
```suggestion
void load(std::unique_ptr<Cargo> cargo) {
        cargo_.push_back(std::move(cargo));}
```

If you would like this to be done by unique_ptr

Reply:
I'd rather stick with share_ptr for now, but it's just my opinion. I don't know how much it would later differ in usage if we now choose shared or unique?

Reply:
I'm not pushing for uniques. Just a suggestion.

Main thing i think would be the need to move pointer if you wanted to take it outside of class for any reason and change it's "owner". And I can't think of any situation like that.